fsspec/parquet.py

```diff
@@ -54,10 +65,19 @@ def open_parquet_file(
     columns: list, optional
         List of all column names that may be read from the file.
     row_groups : list, optional
-        List of all row-group indices that may be read from the file.
+        List of all row-groups that may be read from the file. This
+        may be a list of row-group indices (integers), or it may be
+        a list of `RowGroup` metadata objects (if the "fastparquet"
+        engine is used).
     storage_options : dict, optional
         Used to generate an `AbstractFileSystem` object if `fs` was
         not specified.
+    strict : bool, optional
+        Whether the resulting `KnownPartsOfAFile` cache should
+        fetch reads that go beyond a known byte-range boundary.
+        If `False` (the default), any read that ends outside a
+        known part will be zero padded. Note that using
+        `strict=True` may be useful for debugging.
```

Comment:
(should have a logger in KnownPartsCache?)

Reply:
Good call - Added logger when an "unknown" range is fetched